packages/app/src/systems/Core/utils/math.ts

```diff
@@ -61,7 +61,9 @@ export function parseToFormattedNumber(
   if (typeof value === 'number') {
     val = BigInt(Math.trunc(value));
   }
-  return ethers.commify(toFixed(formatUnits(val, precision), FIXED_UNITS));
+  const result = ethers.commify(toFixed(formatUnits(val, precision), FIXED_UNITS));
+  const decimal = new Decimal(result.replaceAll(',', ''));
+  return decimal.lessThanOrEqualTo('0.0001') ? '0' : result;
```

Comment:
Why we need this here?

Reply:
Yes, is returning 100% when the user has 100% of the pool and 0 when he have nothing.

Reply:
what do you recommend doing in this case @luizstacio?

Reply:
let's remove the "hack" implementation and address it on the gas fee in tx. For let the issue open.

Reply:
✅